File: go/structural/Composite/OneTeaBag.go
package composite


//OneTeaBag - one composite extension - the "leaf"

import java.util.ListIterator

public class OneTeaBag extends TeaBags { 
    public OneTeaBag(string nameIn) {
        this.setName(nameIn)
    }
    
    public int countTeaBags() {
        return 1
    }
   
    public boolean add(TeaBags teaBagsToAdd) {
        return false
    }
    public boolean remove(TeaBags teaBagsToRemove) {
        return false
    }
    public ListIterator createListIterator() {
        return null
    }
}

File: scala/src/main/scala/Structural/Composite/OneTeaBag.scala
package Structural.Composite;


//OneTeaBag - one composite extension - the "leaf"

import java.util.ListIterator;

class OneTeaBag extends TeaBags { 
    OneTeaBag(String nameIn) {
        this.setName(nameIn);
    }
    
    int countTeaBags() {
        return 1;
    }
   
    boolean add(TeaBags teaBagsToAdd) {
        return false;
    }
    boolean remove(TeaBags teaBagsToRemove) {
        return false;
    }
    ListIterator createListIterator() {
        return null;
    }
}